USAGE.md

```diff
@@ -0,0 +1,142 @@
+
+- If you are coding in Angular 1.4 style, you can read 
+[Best Practice Recommendations for Angular App Structure](https://docs.google.com/document/d/1XXMvReO8-Awi1EZXAXS4PzDzdNvV6pGcuaF4Q9821Es/pub). 
+- Concerning Angular ≥ 1.5 and Angular 2, you can read [Angular 2 Style Guide](https://mgechev.github.io/angular2-style-guide/). 
+- For React, you can read [Airbnb React Style Guide](https://github.com/airbnb/javascript/tree/master/react).
+
+<pre>
+├──  conf/
+│    ├──  browsersync-dist.conf.js
+│    ├──  browsersync.conf.js
+│    ├──  gulp.conf.js
+│    ├──  karma-auto.conf.js
+│    ├──  karma.conf.js
+│    ├──  ts.conf.json
+│    ├──  tslint.conf.json
+│    ├──  webpack.conf.js
+│    ├──  webpack-dist.conf.js
+│    ├──  webpack-test.conf.js
+│    └──  karma.conf.js
+│
+├──  gulp_tasks/
+│    ├──  browsersync.js
+│    ├──  build.js
+│    ├──  inject.js (optional)
+│    ├──  karma.js
+│    ├──  misc.js
+│    ├──  scripts.js
+│    ├──  styles.js
+│    ├──  systemjs.js (optional)
+│    └──  webpack.js (optional)
+│
+├──  bower_components/ (optional)
+├──  jspm_packages/ (optional)
+├──  nodes_modules/
+├──  typings/ (optional)
+│
+├──  src/
+│   ├──  app/
+│   │   ├── techs/
+│   │   │   ├──  tech.(js|ts)
+│   │   │   ├──  tech.spec.(js|ts)
+│   │   │   ├──  techs.(js|ts)
+│   │   │   └──  techs.spec.(js|ts)
+│   │   │
+│   │   ├── footer.(js|ts)
+│   │   ├── footer.spec.(js|ts)
+│   │   ├── header.(js|ts)
+│   │   ├── header.spec.(js|ts)
+│   │   ├── main.(js|ts)
+│   │   ├── main.spec.(js|ts)
+│   │   ├── title.(js|ts)
+│   │   └── title.spec.(js|ts)
+│   │
+│   ├── index.html
+│   ├── index.(js|ts)
+│   └── index.(css|less|scss)
+│
+├──  .babelrc (optional)
+├──  .editorconfig
+├──  .gitignore
+├──  .eslintrc
+├──  bower.json (optional)
+├──  gulpfile.js
+├──  jspm.config.js (optional)
+├──  jspm.browser.js (optional)
+└──  package.json
+</pre>
+
+## Configuration
+
+You can file the configuration files in `conf/`.
+
+* *browsersync.conf(-dist).js*: define which folder to serve
+* *gulp.conf.js*: contains the variables used in other gulp files which defines tasks
+* *karma(-auto).conf.js*: contains Karma configuration for unit tests
+* *ts.conf.js*: TypeScript compiler configuration regarding the framework used
+* *tslint.conf.json*: TypeScript linter rules
+* *webpack(-dist | test).js*: Webpack configuration with definition of loaders and plugins regarding the environment (dev, dist or test)
+
+## Gulpfile
+
+
+
+## Questions the generator will ask
+
+- *Framework*: React, Angular 1, Angular 2
+- *Modules manager*: Webpack with NPM, SystemJS with JSPM, None with Bower injection
+- *CSS preprocessor*: SASS, Less, CSS
+- *JS preprocessor*: ES2015 with Babel, Pure old JavaScript, TypeScript
+- *Sample app*: A working landing page, A simple Hello Word, a Redux TodoApp
```

Comment:
I would prefer Boilerplate instead of Sample app like html5-boilerplate, https://github.com/sindresorhus/electron-boilerplate etc